Pullupstream/v0.5.0 rc4

arnaubennassar opened the pull request:
Instead of reviewing this PR code diff, which is absolutely insane, review the code diff vs zkevm-node. Instructions on `docs/diff/diff.md` included on this branch

dependabot[bot] committed:
build(deps): bump golang.org/x/sync from 0.4.0 to 0.5.0 (#2755)

Bumps [golang.org/x/sync](https://github.com/golang/sync) from 0.4.0 to 0.5.0.
- [Commits](https://github.com/golang/sync/compare/v0.4.0...v0.5.0)

---
updated-dependencies:
- dependency-name: golang.org/x/sync
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump github.com/gorilla/websocket from 1.5.0 to 1.5.1 (#2754)

Bumps [github.com/gorilla/websocket](https://github.com/gorilla/websocket) from 1.5.0 to 1.5.1.
- [Release notes](https://github.com/gorilla/websocket/releases)
- [Commits](https://github.com/gorilla/websocket/compare/v1.5.0...v1.5.1)

---
updated-dependencies:
- dependency-name: github.com/gorilla/websocket
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump github.com/fatih/color from 1.15.0 to 1.16.0 (#2758)

Bumps [github.com/fatih/color](https://github.com/fatih/color) from 1.15.0 to 1.16.0.
- [Release notes](https://github.com/fatih/color/releases)
- [Commits](https://github.com/fatih/color/compare/v1.15.0...v1.16.0)

---
updated-dependencies:
- dependency-name: github.com/fatih/color
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

agnusmor committed:
Cherry pick effective gas prices fixes (#2738, #2741) to develop branch (#2766)

* Fix GlobalQueue/AccountQueue pool config (#2738)

* Fix missing pool.Config parameter when creating a new Sequencer (#2741)

* add gasprice update log

* log finalizer L1/L2 gas prices

* fix passing pool.Config to new sequencer

* remove temporary logs

joanestebanr committed:
Fix e2e group 10 (forced batches) (#2750)

* adapt ARR552 fixes from PR #2694
* add failfast to reduce the time on fail
* fix problem with update GER
* avoid duplicated code for vector_groupx

joanestebanr committed:
+ renamed config vars for L1 parallel sync (cherry-pick from #2768) (#2772)

* + renamed config vars for L1 parallel sync

dependabot[bot] committed:
build(deps): bump golang.org/x/net from 0.17.0 to 0.18.0 (#2770)

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.17.0 to 0.18.0.
- [Commits](https://github.com/golang/net/compare/v0.17.0...v0.18.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ToniRamirezM committed:
Refactor State: Step 1  (#2773)

* state refactor

* enable tests

* fix tests

joanestebanr committed:
Fix e2e group 8 (debug trace)  (#2761)

* fix problem if L1 block have 2 transactions

tclemos committed:
add queue to ws subscription filters (#2769) (#2779)

ToniRamirezM committed:
Update GER on closed batch without txs (#2793) (#2794)

* update GER on closed batch without txs

* refactor

ToniRamirezM committed:
Fix data stream tool performance (#2785)

* improve tool performance

* update streamer lib

* fix

* fix

* fix query

* fix

* update streamer lib

* remove unused code

* fix makefile text

* fix makefile text

ARR552 committed:
L1InfoTree implementation + tests (#2786)

* implementation + tests

* linter

* Public HashLeafData

jsvisa committed:
etherman: read tx with blockHash and txIndex instead of txHash (#2722)

* etherman: read tx with blockHash and txIndex instead of txHash

Signed-off-by: jsvisa <delweng@gmail.com>

* etherman: use TransactionInBlock instead

Signed-off-by: jsvisa <delweng@gmail.com>

---------

Signed-off-by: jsvisa <delweng@gmail.com>

ToniRamirezM committed:
Fix tx size check (#2799) (#2804)

* fix tx size check

* fix tx size check

* refactor

* fix test

* fix test

* fix

agnusmor committed:
Added L2GasPriceSuggesterFactor config parameter to simulate estimated L2 gas price (EGP disabled) (#2777) (#2805)

* added L2GasPriceSuggesterFactor config parameter to "simulate" estimated l2 gas price when egp is disabled

* fix L2GasPriceSuggesterFactor value in test config file

* fix L2GasPriceSuggesterFactor value in test config file

* update doc

omahs committed:
Fix typos (#2803)

joanestebanr committed:
Refactor incaberry L1 event executors to another files to avoid modificate it when implement ETROG fork (#2812)

* Isolate L1 event code in classes

Co-authored-by: Alonso <ARR551@protonmail.com>

ARR552 committed:
compute merkleproof (#2820)

* merkleproof

* json doc

agnusmor committed:
Cherry-pick #2658: feature: storing LER in state.batch on successful transaction processing (#2821)

* feature: storing LER in state.batch on successful transaction processing (#2658)

Signed-off-by: Nikolay Nedkov <nikolai_nedkov@yahoo.com>

* fix doc

---------

Signed-off-by: Nikolay Nedkov <nikolai_nedkov@yahoo.com>
Co-authored-by: Nikolay Nedkov <nikolai_nedkov@yahoo.com>

ARR552 committed:
Fix timestamp stored in exit_root table (#2811)

* Fix timestamp stored in exit_root table

* json doc

dependabot[bot] committed:
build(deps): bump github.com/0xPolygonHermez/zkevm-data-streamer (#2824)

Bumps [github.com/0xPolygonHermez/zkevm-data-streamer](https://github.com/0xPolygonHermez/zkevm-data-streamer) from 0.1.11 to 0.1.13.
- [Release notes](https://github.com/0xPolygonHermez/zkevm-data-streamer/releases)
- [Commits](https://github.com/0xPolygonHermez/zkevm-data-streamer/compare/v0.1.11...v0.1.13)

---
updated-dependencies:
- dependency-name: github.com/0xPolygonHermez/zkevm-data-streamer
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ARR552 committed:
#2669 LxLy bridge support (#2694)

* smc

* rename matic to pol

* WIP

* Etherman + unit tests

* Addresses

* aggregator

* New genesis file

* new genesis

* state unit tests

* aggregator unit test

* sequencer uni tests

* pool unit test

* fix symLinks

* linter

* documentation

* Wip

* oldPolygonZkevmBridge

* fix e2e group 1

* linter

* docs

* Fix init Batch 1

* linter

* Fix e2e

* Fix forced

* Revert "Fix forced"

This reverts commit cfaa28bfb90d6e808fd43502867ef7418ca58f46.

* wip

* fix forced e2e

* support genesis config with pointer to BatchData (#2729)

* genesis fix

* forkID 6

* fix e2e forcedBatches again

* new smc

* genesis

* linter

* Fix e2e

* genesis

* new docker + genesis

* json doc

* fix e2e

tclemos committed:
new tracers (#2663)

tclemos committed:
add response headers to all http responses (#2829) (#2830)

joanestebanr committed:
fork-etrog: synchronizer: update L1InfoTree (#2818)

* update LxLy changes


---------

Co-authored-by: Alonso Rodriguez <ARR552@users.noreply.github.com>

joanestebanr committed:
synchronized refactor: move l1_parallel to a folder (#2836)

* moved l1_parallel files to l1_parallel_sync folder

joanestebanr committed:
fix e2e group 5 (#2838)

* same solution as group 8

ToniRamirezM committed:
etrog Integration with executor (#2827)

* wip

* new tracers

* update prover image

* update prover config

* remove unused method

* converters v2

* process batch v2

* update logs

* new errors

* refactor

* doc

* new zk counter

* update prover image

* refactor response v1 to look like v2

* fix test

* change prover image

* comment test

* comment test

* disable check superset batch transactions

* fix store txs

* fix store txs

* fix store txs

* fix store txs

* uncomment test

* change handleForcedTxsProcessResp

* change handleForcedTxsProcessResp

* fix ExecuteBatchV2

* revert executor image

* add l2 tx hash to process tx response

* udpate hashdb proto (#2833)

* Refactor finalizer: remove use of f.processRequest, move getLastBatchNumAndOldStateRoot function (#2834)

* remove f.processRequest. Move getLastBatchNumAndOldStateRoot to sequencer

* refactor get wip batch when starting sequencer/finalizer

* remove consts for numbers

* Refactor finalizer: remove sharedResourcesMux, reduce logs with info level (#2835)

* remove f.processRequest. Move getLastBatchNumAndOldStateRoot to sequencer

* refactor get wip batch when starting sequencer/finalizer

* remove sharedResourcesMux. reduce logs with info level

---------

Co-authored-by: agnusmor <100322135+agnusmor@users.noreply.github.com>

joanestebanr committed:
etrog: L1InfoTreeIndex is 32bits (#2844)

* L1InfoTreeIndex is 32bits (new type L1InfoTreeIndexType)

dependabot[bot] committed:
build(deps): bump github.com/go-git/go-git/v5 from 5.10.0 to 5.10.1 (#2850)

Bumps [github.com/go-git/go-git/v5](https://github.com/go-git/go-git) from 5.10.0 to 5.10.1.
- [Release notes](https://github.com/go-git/go-git/releases)
- [Commits](https://github.com/go-git/go-git/compare/v5.10.0...v5.10.1)

---
updated-dependencies:
- dependency-name: github.com/go-git/go-git/v5
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump golang.org/x/crypto from 0.15.0 to 0.16.0 (#2851)

Bumps [golang.org/x/crypto](https://github.com/golang/crypto) from 0.15.0 to 0.16.0.
- [Commits](https://github.com/golang/crypto/compare/v0.15.0...v0.16.0)

---
updated-dependencies:
- dependency-name: golang.org/x/crypto
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump github.com/holiman/uint256 from 1.2.3 to 1.2.4 (#2849)

Bumps [github.com/holiman/uint256](https://github.com/holiman/uint256) from 1.2.3 to 1.2.4.
- [Release notes](https://github.com/holiman/uint256/releases)
- [Commits](https://github.com/holiman/uint256/compare/v1.2.3...v1.2.4)

---
updated-dependencies:
- dependency-name: github.com/holiman/uint256
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump golang.org/x/net from 0.18.0 to 0.19.0 (#2852)

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.18.0 to 0.19.0.
- [Commits](https://github.com/golang/net/compare/v0.18.0...v0.19.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ToniRamirezM committed:
expose SkipFirstChangeL2Block (#2853)

dependabot[bot] committed:
build(deps): bump github.com/spf13/afero from 1.10.0 to 1.11.0 (#2859)

Bumps [github.com/spf13/afero](https://github.com/spf13/afero) from 1.10.0 to 1.11.0.
- [Release notes](https://github.com/spf13/afero/releases)
- [Commits](https://github.com/spf13/afero/compare/v1.10.0...v1.11.0)

---
updated-dependencies:
- dependency-name: github.com/spf13/afero
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

joanestebanr committed:
fix camelCase to snake_function that fails in MAC (#2862)

ToniRamirezM committed:
compute wip batch during stream delta calculation (#2864)

ToniRamirezM committed:
Compute and store l2 tx hash (#2855)

* WIP

* l2 hash

* add tests

* linter

* fix sql

* fix test

* fix no sender

* fix no sender

* fix test

* fix test

dependabot[bot] committed:
build(deps): bump github.com/0xPolygonHermez/zkevm-data-streamer (#2870)

Bumps [github.com/0xPolygonHermez/zkevm-data-streamer](https://github.com/0xPolygonHermez/zkevm-data-streamer) from 0.1.13 to 0.1.14.
- [Release notes](https://github.com/0xPolygonHermez/zkevm-data-streamer/releases)
- [Commits](https://github.com/0xPolygonHermez/zkevm-data-streamer/compare/v0.1.13...v0.1.14)

---
updated-dependencies:
- dependency-name: github.com/0xPolygonHermez/zkevm-data-streamer
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

joanestebanr committed:
etrog: Decode/coding etrog batches (#2854)

* encoding/decoding etrog batches

ToniRamirezM committed:
update executor proto (#2872)

* update executor proto

* update doc

ARR552 committed:
Feature/#2831 new smc etrog (#2837)

* wip

* linter

* docker + genesis

* new prover image

* docker image v4.0.0-RC2

* Feature/#2840 l1 info root (#2865)

* Keep siblings in memory + AddLeaf + tests

* l1infotree state and synchronizer

* linter

* fix: clead db before run the test

* index type uint32

* prover image v4.0.0-RC3

* ForcedBatch fields added

tclemos committed:
change state to control the new hash for ETROG blocks (#2845)

tclemos committed:
improve OOC and OOG errors (#2774)

tclemos committed:
cherry pick the latest WS improvements into develop (#2819)

agnusmor committed:
Add StoreL2Block (ETROG) function to the state (#2875)

* add StoreL2Block function to the state

* fix linter

* fix linter

agnusmor committed:
Sequencer Etrog refactor (#2878)

* sequencer etrog draft 1

* refactor sequencer etrog implemetation

ToniRamirezM committed:
add im state root to data stream (#2893) (#2895)

* add im state root to data stream

* fix test

* remove log

* optimization

* fix config

* init map

* add wg

* add logs

* add logs

* add logs

* add logs

* add logs

* add logs

* fix

tclemos committed:
review Estimate Gas validations and Formulas (#2856) (#2897)

tclemos committed:
jRPC - Review Logs (#2883) (#2899)

ToniRamirezM committed:
fix test (#2900)

joanestebanr committed:
wip: etrog: synchronize trusted batches (#2871)

* docker + genesis
* new prover image
* wip: isolate syncTrustedState code
* Feature/#2840 l1 info root (#2865)
* Keep siblings in memory + AddLeaf + tests
* l1infotree state and synchronizer
* synchronizer:moved common objects to common package
* pre-execution trs use ETROG
* fix generateReceipt. Use StoreL2Block from the state (#2880)
* refactor init/get wip batch. Add SHA256 counter. Fixes (#2882)
* process etrog batches with ExecuteBatchV2
* Fix final StateRoot for batch when it is closed (#2887)
* removed timestamp comparation on virtualization of batches
* timestamp check
* use time.Now() as minTimestamp for open a trusted batch

---------

Co-authored-by: Alonso <ARR551@protonmail.com>
Co-authored-by: Alonso Rodriguez <ARR552@users.noreply.github.com>
Co-authored-by: agnusmor <100322135+agnusmor@users.noreply.github.com>
Co-authored-by: tclemos <thiago@polygon.technology>
Co-authored-by: Toni Ramírez <58293609+ToniRamirezM@users.noreply.github.com>
Co-authored-by: agnusmor <agnusmor@gmail.com>

dpunish3r committed:
WIP

ToniRamirezM committed:
fix ProcessBatchV2 (#2905)

* fix ProcessBatchV2

* fix ProcessBatchV2

* fix ProcessBatchV2

* update prover image

ARR552 committed:
update prover version v4.0.0-RC8 (#2908)

dpunish3r committed:
WIP

ARR552 committed:
l1InfoRoot + fixes unsigned tx (#2911)

* l1InfoRoot + fixes unsigned tx

* fix linter

dpunish3r committed:
WIP

joanestebanr committed:
etrog: batch timestamp is set when is virtualized (#2904)

* + add field virtual.timestamp_batch_etrog into DB, rpc returns this timestamp of the batch of null if not defined yet

ToniRamirezM committed:
state transtiion v2 test (#2912)

* state transtiion v2 test

* conflicts

* remove comments

ToniRamirezM committed:
fix forkID in ProcessRequest (#2915)

joanestebanr committed:
Feature/2914 etrog use l1inforoot from event (#2918)

* + use L1inforoot, set forkid in process trusted batch

* removed comment

agnusmor committed:
fix update last L1InfoRoot (#2921)

agnusmor committed:
update/get batch used resources from the state (batch table) (#2916)

ToniRamirezM committed:
Refactor state tests (#2920)

* refactor state tests

* refactor state tests

dpunish3r committed:
WIP

tclemos committed:
review Estimate Gas validations and Formulas (#2856) (#2876)

tclemos committed:
review usigned tx execution and make estimate gas compatible with all forkIDs (#2922)

agnusmor committed:
fix tx size when calculating egp and checking remaining batch resources (#2924)

joanestebanr committed:
etrog: support permissionless sync from scratch (#2923)

* skip trusted batch until 2, fix error OOC report
* renamed some classes

dpunish3r committed:
Bugs fixed

ARR552 committed:
Fix/l1 info root (#2925)

* fixes L1InfoRoot

* more fixes

dpunish3r committed:
Improve display stats. Add README.

ToniRamirezM committed:
Feature/state tests refactor2 (#2928)

* refactor state tests

* refactor state tests

* refactor state tests

* close state

* conflicts

dpunish3r committed:
Config for testnet

joanestebanr committed:
etrog: use WIP  flag in the response of zkevm RPC for Closed field, add WIP to state.Batch (#2927)

* add WIP to state.batch struct
* Change how to determine is a batch is closed in jsonrpc
* missing wip field in some sql

dependabot[bot] committed:
build(deps): bump github.com/google/uuid from 1.4.0 to 1.5.0 (#2902)

Bumps [github.com/google/uuid](https://github.com/google/uuid) from 1.4.0 to 1.5.0.
- [Release notes](https://github.com/google/uuid/releases)
- [Changelog](https://github.com/google/uuid/blob/master/CHANGELOG.md)
- [Commits](https://github.com/google/uuid/compare/v1.4.0...v1.5.0)

---
updated-dependencies:
- dependency-name: github.com/google/uuid
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump google.golang.org/grpc from 1.59.0 to 1.60.0 (#2894)

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.59.0 to 1.60.0.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.59.0...v1.60.0)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump github.com/urfave/cli/v2 from 2.25.7 to 2.26.0 (#2874)

Bumps [github.com/urfave/cli/v2](https://github.com/urfave/cli) from 2.25.7 to 2.26.0.
- [Release notes](https://github.com/urfave/cli/releases)
- [Changelog](https://github.com/urfave/cli/blob/main/docs/CHANGELOG.md)
- [Commits](https://github.com/urfave/cli/compare/v2.25.7...v2.26.0)

---
updated-dependencies:
- dependency-name: github.com/urfave/cli/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump github.com/go-git/go-git/v5 from 5.10.1 to 5.11.0 (#2888)

Bumps [github.com/go-git/go-git/v5](https://github.com/go-git/go-git) from 5.10.1 to 5.11.0.
- [Release notes](https://github.com/go-git/go-git/releases)
- [Commits](https://github.com/go-git/go-git/compare/v5.10.1...v5.11.0)

---
updated-dependencies:
- dependency-name: github.com/go-git/go-git/v5
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump github.com/spf13/viper from 1.17.0 to 1.18.1 (#2889)

Bumps [github.com/spf13/viper](https://github.com/spf13/viper) from 1.17.0 to 1.18.1.
- [Release notes](https://github.com/spf13/viper/releases)
- [Commits](https://github.com/spf13/viper/compare/v1.17.0...v1.18.1)

---
updated-dependencies:
- dependency-name: github.com/spf13/viper
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

agnusmor committed:
fix TODOs. fix reprocessFullBatch. fix egp tx size (#2929)

* fix TODOs. fix reprocessFullBatch. fix egp tx size

* fix linter

* fix L1InfoTree_V2 on reprocessFullBatch

* fix egp tests

ToniRamirezM committed:
New state test vectors (#2933)

* new state test vectors

* remove log

ToniRamirezM committed:
Revert "build(deps): bump github.com/spf13/viper from 1.17.0 to 1.18.1 (#2889)" (#2935)

This reverts commit f4885b3aa2837cf6aa0a824cfc81f581601d6265.

dpunish3r committed:
Add stat gas price before EGP. Add l2 block timestamp

ARR552 committed:
Update smc + sync init tx (#2941)

* new smc

* wip sync init tx

* sync init tx

* fix

* linter

* json

* new image and genesis

tclemos committed:
extend block endpoints to provide the new l2 block information (#2936)

ToniRamirezM committed:
handle new executor errors (#2942)

dpunish3r committed:
Add stats tx fee. New config file

ToniRamirezM committed:
expose new V2 executor fields (#2944)

ToniRamirezM committed:
Update aggregator for etrog (#2943)

* wip

* update aggregator for etrog

* remove TODO

ToniRamirezM committed:
fix genesis sequence (#2950)

dpunish3r committed:
Bug fix uint64 overflow. Add showalways option

agnusmor committed:
Refactor forced batches (ETROG). Fix/Review executor requests (#2946)

* etrog refactor forced batches. fix executor requests

* update etrog state test

* fix state test

* remove log

* fix tool

* update prover image

---------

Co-authored-by: Toni Ramírez <toni@polygon.technology>

ToniRamirezM committed:
Fix/none2e tests (#2955)

* network test

* default test

* default test

* comment etherman tests

* uncomment etherman test

* remove error

agnusmor committed:
return last L1InfoRoot in function GetL1InfoTreeDataFromBatchL2Data (#2956)

tclemos committed:
reviews l2 block exit roots and exposes endpoint to get them by ger (#2957)

ARR552 committed:
Feature/#2948 v1tov2 (#2958)

* make, docker compose and genesis

* fix init tx

* fix event

* Fix genesis

* fix process forced batch

dpunish3r committed:
Remove conditions. Update config

joanestebanr committed:
etrog: synchornizer-permisionless: first version working of sync with trusted state (#2949)

- state: Changed `ProcessingContextV2` to support `L1InfoTreeData` (all the leafs of L1InfoTree used in a batch)
- state: removed debug logs from the `enconding_batch_v2.go`
- test: upgrade Mockery from 2.22.1 to 2.39.0 to support `max` function of golang 1.21
- synchronizer: tidy up interfaces. The repeated have been grouped
- synchronizer: simplified generation of mocks (stop using "mock_" prefix but are under `mocks/` folder)
- synchronizer: change cache system of syncTrusted
- synchronizer: to known if a trusted batch is closed  stop using empty StateRoot and use `Closed` field
- synchronizer: don't check batch timestamp because RPC is not providing the creational_stamp (issue #2953
- synchronizer: calls to executor pass the `L1InfoTreeData`

ARR552 committed:
Fix UUID + prover rc16 (#2963)

agnusmor committed:
remove accInputHash calculation. Fix batch GER (#2959)

joanestebanr committed:
use prover 4.0.0-RC16 for permissionless (#2964)

ToniRamirezM committed:
fix mockery (#2961)

* fix mockery

* fix mockery

* update prover image

* update prover image

* doc

ToniRamirezM committed:
fix finalizer l1InfoTreeData (#2965)

* fix finalizer l1InfoTreeData

* fix finalizer l1InfoTreeData

ToniRamirezM committed:
aggregate forced batches (#2968)

* aggregate forced batches

* update comment

* comments

* comments

dpunish3r committed:
Add show simulation only flag. Add show reprocessed info. Update README

threewebcode committed:
docs: refine the syntax and make it more clean and clear. (#2873)

joanestebanr committed:
Feature/etrog support no data on db exit root (#2973)

* leaf with index 0 must not be included in l1InfoTreeData

ToniRamirezM committed:
new executor errors (#2972)

ToniRamirezM committed:
Fix/aggregator decoding (#2971)

* fix aggregator decoding

* fix aggregator decoding

* refactor

estensen committed:
chore: remove unnecessary conversions (#2960)

ToniRamirezM committed:
fix aggregator index 0 (#2975)

agnusmor committed:
Remove closing signal manager. Move dbManager config to Sequencer section (#2974)

* Remove closing signal manager. Remove dbManager config. Fixes

* fix linter and doc

* remove return after f.Halt()

tclemos committed:
add block information to exit roots response (#2970)

tclemos committed:
add support to tx l2 hash for RPC responses (#2945)

ARR552 committed:
fix smc (#2967)

* fix smc

* fix

alexqrid committed:
Added nodes last synced batch number (#2590)

* Added nodes last synced batch number

* Added function doc

---------

Co-authored-by: alexqrid <>

agnusmor committed:
remove unused sequencer config parameters. Remove unnecessary empty l2 block execution (#2978)

dpunish3r committed:
Solve linter issues

dpunish3r committed:
Update datastream lib. Sequencer order batches to send to datastream

joanestebanr committed:
etrog: synchronizer permissionless: Check that permissionless are enough synchronized with L1 to process Trusted Batch (#2985)

* check GlobalExitRoot of trustedBatch if we have it or we are behind the expected L1 sync
* sleep 1 if trustedSync is respawned

dpunish3r committed:
Merge pull request #2988 from 0xPolygonHermez/cherry-pick/dslib-orderbatch

Update datastream lib. Sequencer order batches to send to datastream

agnusmor committed:
fix log batch response when we got an error calling executor (#2990)

tclemos committed:
adapt debug methods to run v1 and v2 executor requests (#2991)

agnusmor committed:
rename finalizer config parameters to be more consistent (#2996)

agnusmor committed:
review logs to be more consistent. fix resource sustract in new wip batch (#2998)

agnusmor committed:
Fix sequencer/executor metrics. Rename poolPackage/statePackage to poolIntf/stateIntf. (#3003)

* fix metrics. Rename poolPackage/statePackage to poolInft/stateInft.

* fix linter (time.Since)

* remove state.transaction .sed_sha256_hashes field from migrations

tclemos committed:
Fix/l2 block nonce (#3008)

ToniRamirezM committed:
Data stream tool improvements (#3006)

* fix decode l2block tool

* im root cache

* im root cache

* refactor

* refactor

* refactor

* refactor

* refactor

* refactor

* refactor

* change comment

* fix log

* fix

* fix

ARR552 committed:
fix (#2982)

agnusmor committed:
Optmize storage.StoreL2Block function. Fix tx index on receipt (#3009)

* optmize storage.StoreL2Block function. Fix tx index on receipt

* Use CopyFrom in storage.storeL2Block related functions

agnusmor committed:
Use %v instead %w to log errors. Fix HaltOnBatchNumber (#3012)

tclemos committed:
e2e test fixes for groups 1, 2, 3, 4 and 7 (#3010)

joanestebanr committed:
fix sync permissionless partial process (#3017)

ARR552 committed:
Fixes executor error handling (#3019)

* wip

* fix handle error +executor  rc 19

agnusmor committed:
Fix process empty L2 block (#3021)

* test l2block sync

* log L1InfoTreeIndex

* remove l2block sync

* remove commented sync l2block code

agnusmor committed:
fix finalizer forced batches. fix storage.GetBatchByForcedBatchNum (#3024)

agnusmor committed:
Disable e2e forced batches tests, groups 9, 10 and 11 (#3025)

* disable e2e tests groups 9, 10 and 11

* disable sequencer UT not working on ETROG

ToniRamirezM committed:
Fix delta stream etrog (#3022)

* fix delta stream for etrog

* fix query for backwards compatibility

* fix query for backwards compatibility

* fix

* add labels to loops

* fix

* fix

* fix

* add sanity check when adding l2blocks

* fix

* fix

* fix

* fix

tclemos committed:
Allow extra l2 information to be retrieved from standard endpoints conditionally (#3029)

ARR552 committed:
Fix/sync l1 info tree data (#3033)

* fix l1InfoTreeData when syncing

* fix

* remove leaves in forcedBatches

agnusmor committed:
Cherry-pick #3000: Add zkevm_EstimateFee endpoint (#3035)

* add zkevm_EstimateFee endpoint (#3000)

* update pool mock

joanestebanr committed:
fix synchronizer unittest and e2e group 6 (#3032)

* fix e2e group 6
* If a L1 virtual batch  detects missmatch with state batch doesn't create a entry in trusted_reorg table if the batch is WIP
* Add new Makefile target to show dockers logs

ARR552 committed:
fixes v1tov2 (#3039)

estensen committed:
chore: unslice slice expressions (#2976)

estensen committed:
chore: replace strings.SplitN(arg, sep, 2) with strings.Cut(arg, sep) (#2977)

AdventureSeeker987 committed:
chore: fix typos in Readme (#2980)

arnaubennassar committed:
Fix conflicts

ToniRamirezM committed:
Add new etrog fields to Data Stream and update tool (#3043)

* new fields in stream

* read blockHashL1

* fix sql query

* fix sql query

* logs

* logs

* logs

* logs

* logs

* logs

* logs

* fix test

* fix linter

* update doc

* fill l1InfoRoot

arnaubennassar committed:
WIP: pending smart contract fix to be able to compile

arnaubennassar committed:
Code compiles

dpunish3r committed:
wip

ARR552 committed:
Fix/aggregator etrog inforoot (#3048)

* fix inforoot

* fixes

---------

Co-authored-by: Toni Ramírez <toni@polygon.technology>

arnaubennassar committed:
Update L1 docker image

ARR552 committed:
fix (#3049)

tclemos committed:
revert l2 block hash to ethereum standard (#3045)

arnaubennassar committed:
Move DAC to separate package, code diff focused on etherman instead of sync

arnaubennassar committed:
reduce code diff

arnaubennassar committed:
dacMan WIP

dpunish3r committed:
wip

dpunish3r committed:
wip

ARR552 committed:
fix l1InfoRoot in sync, aggregator and state (#3052)

* fix l1InfoRoot in sync, aggregator and state

* fix

joanestebanr committed:
Add more unittest to synchronizer (#3040)

* add more unittest for trusted sync 
* new uniitest for trusted sync and control a case of not able to close a batch because the data doesnt match
* fix batch that dont need process but could closed  a batch
* fix status update
* fix desync problem between producer and consumer

arnaubennassar committed:
Modular DA

arnaubennassar committed:
Add default DA backend for the config file

dpunish3r committed:
merge develop

arnaubennassar committed:
Create DA backend logic

arnaubennassar committed:
Create DA backend logic

arnaubennassar committed:
A bit of cleaning

dpunish3r committed:
Doc gen

ARR552 committed:
fix (#3055)

joanestebanr committed:
fix error reading RawTxsData from DB for forced Batches (#3056)

dpunish3r committed:
Clean code

arnaubennassar committed:
Lint

arnaubennassar committed:
Only requires equence sender private key for the sequence sender

dpunish3r committed:
Handle unkown type

arnaubennassar committed:
Fix config test

arnaubennassar committed:
WIP fixing UTs

dpunish3r committed:
linter

dpunish3r committed:
Merge pull request #3054 from 0xPolygonHermez/feature/stream-batch-bookmark

Feature/stream batch bookmark

ToniRamirezM committed:
Merge branch 'develop' into tool/egp

dpunish3r committed:
Merge pull request #2986 from 0xPolygonHermez/tool/egp

Tool/egp

ARR552 committed:
fix timestamp (#3061)

ToniRamirezM committed:
remove l1iforoot from stream (#3064)

ARR552 committed:
fix unit tests (#3063)

* fix aggregator unit test

* linter

* synchronizer unit test

* fix AddVirtualBatch unit test

joanestebanr committed:
fix storing GER from L1 events (#3062)

* fix storing GER from L1 events
* fix GER for trusted sync

arnaubennassar committed:
Fix Etherman UTs

arnaubennassar committed:
Fix Etherman UTs

joanestebanr committed:
remove unused code: ZeroL1InfoRoot cte (#3065)

joanestebanr committed:
trusted sync: fix local_exit_root (#3067)

arnaubennassar committed:
WIP fixing UTs

ARR552 committed:
v1 to v2 working (#3069)

* v1 to v2 working

* fix make file

agnusmor committed:
added egp value for eth transfer. Fix zkevm_EstimateFee endpoint. Remove use minGasPriceAllowed on egp calculation (#3041)

Co-authored-by: tclemos <thiago@polygon.technology>

joanestebanr committed:
if a batch have no l2block it returns an zero l1inforoot (#3070)

ToniRamirezM committed:
add new executor errors (#3077)

* add new executor errors

* fix linter

agnusmor committed:
Add L1 block timestamp margin to send sequences (#3078)

* add L1 block timestamp margin to send sequences

* consider negative margin between timestamp of L1 block and last L2 block in the sequence. Fix close empty batch

* fix doc

* update l2block used resources

* add egp UT for breakevengasprice equal to 0

agnusmor committed:
fix finalize l2block when closing batch due to exhausted resource (#3084)

joanestebanr committed:
reprocess trusted batch, use GER of leaf 0 of L1InfoRoot (#3086)

* reprocess trusted batch
* use GER of leaf 0 of L1InfoRoot
* moved hash function to test.helper as HashByteArray
* fix sequencer UT

---------

Co-authored-by: agnusmor <agnusmor@gmail.com>

arnaubennassar committed:
Update contracts and pass first E2E test

tclemos committed:
fix default value for fromBlock when calling eth_getLogs (#3051)

joethechicken committed:
Security - Add SHA Pin to Third Party Actions/Checkout Action (fixes #3001) (#3002)

arnaubennassar committed:
Get DA addr and name from L1

joanestebanr committed:
for L1 sequence batch event if the batch only use leaf 0 set GER as 0x0000..00 (#3092)

arnaubennassar committed:
WIP

arnaubennassar committed:
Add code diff docs

agnusmor committed:
Return zero GER when L1IntroTreeIndex is 0 (#3090)

* return zero GER when L1IntroTreeIndex is 0

* refactor

* refactor

---------

Co-authored-by: Toni Ramírez <toni@polygon.technology>

ToniRamirezM committed:
fix etrog stream generation (#3094)

arnaubennassar committed:
Fix linter

arnaubennassar committed:
Updat config docs

arnaubennassar committed:
Pass DAC test

joanestebanr committed:
synchronizer: remove fatals (#3093)

* removed fatals

ARR552 committed:
Feature/#3081 smc (#3097)

* new contracts

* Event etherman implementation

* smc

* proxy + etherman unit test

* new event implemented in synchronizer

* Fix setforcedBatch event

* smc update

* updateEtrogSequence working

* fix ger mismatch

* add comment suggested

joanestebanr committed:
etrog: Fix synchronizer  OOC sequenced batch (#3095)

* fix. A sequenced batch with OOC error is not a reorg

ToniRamirezM committed:
fix OOC errors (#3100)

* fix OOC errors

* fix OOC errors

tclemos committed:
remove l2 hash from eth_getTransactionReceipt (#3089)

agnusmor committed:
Fix set l1InfoRootIndex on changeL2Block (#3099)

* fix set l1InfoRootIndex on changeL2Block

* increase poseidon hashes used by l2block (temp workaround)

* fix linter/build

* update prover image to v4.0.0-RC27

* fix finalizer tests

Stefan-Ethernal commented on docs/diff/diff.md:
```diff
@@ -0,0 +1,57 @@
+# Diff
+
+This repo is a fork from [zkevm-node](https://github.com/0xPolygonHermez.zkevm-node). The puirpose of the fork is to implement tha Validium consensus, enabling data availability to be posted outside of L1.
```
Minor typos:

```suggestion
This repo is a fork from [zkevm-node](https://github.com/0xPolygonHermez.zkevm-node). The purpose of the fork is to implement the Validium consensus, enabling data availability to be posted outside of L1.
```

ToniRamirezM committed:
fix stream (#3101)

dependabot[bot] committed:
build(deps): bump google.golang.org/protobuf from 1.31.0 to 1.32.0 (#2992)

Bumps google.golang.org/protobuf from 1.31.0 to 1.32.0.

---
updated-dependencies:
- dependency-name: google.golang.org/protobuf
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump github.com/rubenv/sql-migrate from 1.5.2 to 1.6.1 (#3079)

Bumps [github.com/rubenv/sql-migrate](https://github.com/rubenv/sql-migrate) from 1.5.2 to 1.6.1.
- [Commits](https://github.com/rubenv/sql-migrate/compare/v1.5.2...v1.6.1)

---
updated-dependencies:
- dependency-name: github.com/rubenv/sql-migrate
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

agnusmor committed:
Fix batch GER (#3104)

* fix batch GER

* fix use LER

joanestebanr committed:
force version of json-schema-for-humans to 0.47 (#3109)

tclemos committed:
add new endpoint zkevm_getLatestGlobalExitRoot (#3103)

arnaubennassar committed:
Fix config test

arnaubennassar committed:
Minor cleaning

agnusmor committed:
Fix merkle proof in aggregator. Fix finalize batch on sequencer (#3112)

* fix smtProof

* remove logs

* test force l1InfoRootIndex to 0 on startup

* fix close batch due to exhausted resource

* remove force l1InfoRootIndex to 0 on startup

* update prover image to v4.0.0-RC28

* update prover image to v4.0.0-RC29

agnusmor committed:
Fix new wipL2block on finalizeBatch function (#3113)

* fix new wipL2block on finalizeBatch function

* fix sustract wipL2Block usedResources when opening new WIPbatch

* fix openNewWipL2Block

* fix openNewWipL2Block

* update prover image to v4.0.0-RC27

* Update prover image to v4.0.0-RC28

* update prover image to v4.0.0-RC27

Stefan-Ethernal commented on dataavailability/dataavailability.go:
```diff
@@ -0,0 +1,117 @@
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// DataAvailability implements an abstract data avaiolability integration
```
Minor typo:

```suggestion
// DataAvailability implements an abstract data availability integration
```

Stefan-Ethernal commented on dataavailability/dataavailability.go:
```diff
@@ -0,0 +1,117 @@
+			log.Warnf(unexpectedHashTemplate, batchNum, expectedTransactionsHash, actualTransactionsHash)
+		}
+
+		if !d.isTrustedSequencer {
+			log.Info("trying to get data from trusted sequencer")
+			data, err := d.getDataFromTrustedSequencer(batchNum, expectedTransactionsHash)
```
Sorry for the ignorance, but is this logic inverted? In case `isTrustedSequencer == false`, we are fetching data from the trusted sequencer. It looks a bit strange on the first glance, but otoh it might be that if the "local sequencer" is not a trusted one, then retrieve data from the trusted one, right?

Anyways, probably an inline comment would help in understanding this part better.

arnaubennassar replied:
isTrustedSequencer is a config param widely used in the app, kind of a core concept. Basically is saying "I'm the trusted sequencer? yes/no"

It doesnt make sense to pull DA from trusted sequencer if I'm the trusted sequecer already

Stefan-Ethernal commented on dataavailability/dataavailability.go:
```diff
@@ -0,0 +1,117 @@
+
+func (d *DataAvailability) getDataFromTrustedSequencer(batchNum uint64, expectedTransactionsHash common.Hash) ([]byte, error) {
+	b, err := d.zkEVMClient.BatchByNumber(d.ctx, big.NewInt(int64(batchNum)))
+	if err != nil {
```
Is it safe to do down-casting from `uint64` to `int64` here? I'd suggest safer option, wdyt?

```suggestion
	b, err := d.zkEVMClient.BatchByNumber(d.ctx, new(big.Int).SetUint64(batchNum))
```

Stefan-Ethernal commented on dataavailability/datacommittee/datacommittee.go:
```diff
@@ -0,0 +1,292 @@
+// DataCommitteeBackend implements the DAC integration
+type DataCommitteeBackend struct {
+	dataCommitteeContract      *polygondatacommittee.Polygondatacommittee
+	l2Coinbase                 common.Address
```
It is currently unused, are we going to need it in the future, or we can remove it at this point?

Stefan-Ethernal commented on test/scripts/sendForcedBatch/main.go:
```diff
@@ -105,34 +119,34 @@ func sendForcedBatches(cliCtx *cli.Context) error {
-			return err
-		}
-	}
+	// // Allow forced batches in smart contract if disallowed
+	// disallowed, err := zkevm.IsForcedBatchAllowed(&bind.CallOpts{Pending: false})
+	// if err != nil {
+	// 	log.Error("error getting IsForcedBatchAllowed. Error: ", err)
+	// 	return err
+	// }
+	// if disallowed {
+	// 	tx, err := zkevm.ActivateForceBatches(auth)
+	// 	if err != nil {
+	// 		log.Error("error sending activateForceBatches. Error: ", err)
+	// 		return err
+	// 	}
+	// 	err = operations.WaitTxToBeMined(ctx, ethClient, tx, operations.DefaultTimeoutTxToBeMined)
+	// 	if err != nil {
+
+	// 		log.Error("error waiting tx to be mined. Error: ", err)
+	// 		return err
+	// 	}
+	// }
```
Is this going to be needed in the future perhaps?

arnaubennassar replied:
yes, I've diverged from the upstream here due to being ahead in contract integration. IMO it felt better to leave this commented rather than remove test 😅 . But this is going to be fixed in next batch of pulled changed

Stefan-Ethernal commented on etherman/interfaces.go:
```diff
@@ -0,0 +1,7 @@
+
+import "github.com/ethereum/go-ethereum/common"
+
+type dataAvailabilitier interface {
```
`dataAvailabilitier` is hard to spell and even more to pronounce in verbal comunication, that's why I'm suggesting this change if it is still semantically correct :slightly_smiling_face: WDYT?

```suggestion
type l2DataProvider interface {
```

arnaubennassar replied:
agree in that the name isnt great, but since DA is kind of a buzz word in the industry, I think we should follow it. But I can change it to `dataAvailabilityProvider`

Stefan-Ethernal commented on etherman/etherman.go:
```diff
@@ -782,16 +1207,100 @@ func (etherMan *Client) sequencedBatchesEvent(ctx context.Context, vLog types.Lo
+		sequencedBatches := make([]SequencedBatch, len(sequencesValidium))
+		for i, seq := range sequencesValidium {
+			bn := lastBatchNumber - uint64(len(sequencesValidium)-(i+1))
+			batchL2Data, err := da.GetBatchL2Data(bn, sequencesValidium[i].TransactionsHash)
```
`da` field might be `nil`, so wdyt about following null object pattern (providing dummy implementation for `nil` case) and not even care if reference is undefined or not. WDYT?

arnaubennassar replied:
mmmmm, in that context, da should never be nil. In fact the possibility for it to be nil is there cuz of being a fork from something that doesnt need it. I'm ok having a good old panic here if for whatever crazy reason this happens to be nil

Stefan-Ethernal commented on dataavailability/interfaces.go:
```diff
@@ -0,0 +1,25 @@
+	// PostSequence sends the sequence data to the data availability backend, and returns the dataAvailabilityMessage
+	// as expected by the contract
+	PostSequence(ctx context.Context, batchesData [][]byte) ([]byte, error)
+}
```
WDYT about such interface segregation, because if I'm not mistaken, it seems that sometimes you need to implement just the `BatchDataProvider` (reading batches) and sometimes only `SequenceSender` (posting batches, e.g. here https://github.com/0xPolygon/cdk-validium-node/blob/9d186be306740f2fa9e1dcc4186124baa65c6d77/sequencesender/interfaces.go#L42). Does it make sense, or it is an overkill?

```suggestion
// BatchDataProvider is used to retrieve batch data
type BatchDataProvider interface {
	// GetData retrieve the data of a batch from the DA backend. The returned data must be the pre-image of the hash
	GetData(batchNum uint64, hash common.Hash) ([]byte, error)
}

// SequenceSender is used to send provided sequence of batches
type SequenceSender interface {
	// PostSequence sends the sequence data to the data availability backend, and returns the dataAvailabilityMessage
	// as expected by the contract
	PostSequence(ctx context.Context, batchesData [][]byte) ([]byte, error)
}

type DABackender interface {
	BatchDataProvider
	SequenceSender
	// Init initializes the DABackend
	Init() error
}
```

Stefan-Ethernal commented on cmd/run.go:
```diff
@@ -278,12 +288,74 @@ func runMigrations(c db.Config, name string) {
+		)
+		if err != nil {
+			return nil, err
+		}
```
Shall we add default statement and return an error perhaps in order to fail fast, because it would mean we have retrieved some unknown DA protocol name?

ARR552 committed:
fix l1InfoRoot (#3115)

goran-ethernal commented on cmd/run.go:
```diff
@@ -278,12 +288,74 @@ func runMigrations(c db.Config, name string) {
+}
+
+func newDataAvailability(c config.Config, st *state.State, etherman *etherman.Client, isSequenceSender bool) (*dataavailability.DataAvailability, error) {
+	var trustedSequencerURL string
```
```suggestion
	var (
	     trustedSequencerURL string
	     err error
	 )
```

goran-ethernal commented on cmd/run.go:
```diff
@@ -278,12 +288,74 @@ func runMigrations(c db.Config, name string) {
+			log.Debug("getting trusted sequencer URL from smc")
+			trustedSequencerURL, err = etherman.GetTrustedSequencerURL()
+			if err != nil {
+				log.Fatal("error getting trusted sequencer URI. Error: %v", err)
```
I think it's ok to just return an error if it happens in this function, just to be consistent, since on some places in function we do `log.Fatal`, and on some, we return an error. This is called on `newEtherman` function, and on `etherman, err := newEtherman(*c, st)` if an error happens, it does `log.Fatal`. Same happens on `createSequenceSender`.

goran-ethernal commented on cmd/run.go:
```diff
@@ -278,12 +288,74 @@ func runMigrations(c db.Config, name string) {
+	var daBackend dataavailability.DABackender
+	switch daProtocolName {
+	case string(dataavailability.DataAvailabilityCommittee):
+		var pk *ecdsa.PrivateKey
```
```suggestion
		var (
		    pk *ecdsa.PrivateKey
		    err error
		)
```

goran-ethernal commented on cmd/run.go:
```diff
@@ -492,8 +563,20 @@ func newState(ctx context.Context, c *config.Config, l2ChainID uint64, forkIDInt
+	if err != nil {
+		log.Fatal("error getting all leaves. Error: ", err)
+	}
+	var leaves [][32]byte
```
```suggestion
	leaves := make([][32]byte, 0, len(allLeaves))
```

arnaubennassar replied:
this is outside of the diff vs zkevm-node

dependabot[bot] committed:
build(deps): bump google.golang.org/grpc from 1.60.0 to 1.60.1 (#2938)

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.60.0 to 1.60.1.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.60.0...v1.60.1)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

goran-ethernal commented on dataavailability/dataavailability.go:
```diff
@@ -0,0 +1,117 @@
+
+// New creates a DataAvailability instance
+func New(
+	IsTrustedSequencer bool,
```
```suggestion
	IsTrustedSequencer bool,
```
```suggestion
	isTrustedSequencer bool,
```

Stefan-Ethernal commented on cmd/run.go:
```diff
@@ -278,12 +288,74 @@ func runMigrations(c db.Config, name string) {
+func newDataAvailability(c config.Config, st *state.State, etherman *etherman.Client, isSequenceSender bool) (*dataavailability.DataAvailability, error) {
+	var trustedSequencerURL string
+	var err error
+	if !c.IsTrustedSequencer {
```
Nitpick:
```suggestion
	var (
		trustedSequencerURL string
		err                 error
	)
```

Stefan-Ethernal commented on cmd/run.go:
```diff
@@ -278,12 +288,74 @@ func runMigrations(c db.Config, name string) {
+		if c.Synchronizer.TrustedSequencerURL != "" {
+			trustedSequencerURL = c.Synchronizer.TrustedSequencerURL
+		} else {
+			log.Debug("getting trusted sequencer URL from smc")
```
Just curious, what smc abbreviation stands for, is it smart contract?

arnaubennassar replied:
it's just smart contract. IMO SC is way better, but this is inherited from upstream

Stefan-Ethernal commented on cmd/run.go:
```diff
@@ -278,12 +288,74 @@ func runMigrations(c db.Config, name string) {
+	switch daProtocolName {
+	case string(dataavailability.DataAvailabilityCommittee):
+		var pk *ecdsa.PrivateKey
+		var err error
```
```suggestion
		var (
            pk *ecdsa.PrivateKey
		    err error
        )
```

ToniRamirezM committed:
update prover image (#3114)

* update prover image

* update test vectors

dependabot[bot] committed:
build(deps): bump github.com/prometheus/client_golang (#3105)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.17.0 to 1.18.0.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.17.0...v1.18.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

goran-ethernal commented on dataavailability/dataavailability.go:
```diff
@@ -0,0 +1,117 @@
+	}
+	actualTransactionsHash := crypto.Keccak256Hash(transactionsData)
+	if !found || expectedTransactionsHash != actualTransactionsHash {
+		if found {
```
Should we log the `not found` case as well?

arnaubennassar replied:
the reasond behind logging only if found, is because that means that there is a hash missmatch. This implies either a bug or some actor trying to do nasty stuff. But not having found something is expected (for instance when syncing a node from scratch, that log would appear basically once per batch without adding any value)

dependabot[bot] committed:
build(deps): bump golang.org/x/net from 0.19.0 to 0.20.0 (#3106)

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.19.0 to 0.20.0.
- [Commits](https://github.com/golang/net/compare/v0.19.0...v0.20.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

goran-ethernal commented on dataavailability/dataavailability.go:
```diff
@@ -0,0 +1,117 @@
+			log.Info("trying to get data from trusted sequencer")
+			data, err := d.getDataFromTrustedSequencer(batchNum, expectedTransactionsHash)
+			if err != nil {
+				log.Error(err)
```
```suggestion
				log.Error("failed to get data from trusted sequencer: %w", err)
```

goran-ethernal commented on dataavailability/dataavailability.go:
```diff
@@ -0,0 +1,117 @@
+		log.Info("trying to get data from the data availability backend")
+		data, err := d.backend.GetData(batchNum, expectedTransactionsHash)
+		if err != nil {
+			log.Error(err)
```
```suggestion
			log.Error(err)
```
```suggestion
			log.Error("failed to get data from the data availability backend: %w", err)
```

Stefan-Ethernal commented on state/pgstatestorage/pgstatestorage.go:
```diff
@@ -0,0 +1,370 @@
+
+// GetBatchL2DataByNumber returns the batch L2 data of the given batch number.
+func (p *PostgresStorage) GetBatchL2DataByNumber(ctx context.Context, batchNumber uint64, dbTx pgx.Tx) ([]byte, error) {
+	getBatchL2DataByBatchNumber := "SELECT raw_txs_data FROM state.batch WHERE batch_num = $1"
```
Does it make sense to extract this into constant?

arnaubennassar replied:
modified to be a constant. But wee decided to put the queries inside each function. Before we used to have a massive `const` statement with all the sql, and it was painful for the queries and the functions that called the queries to be so separated in space. There is almost an exact 1:1 relation between query and function

goran-ethernal commented on dataavailability/dataavailability.go:
```diff
@@ -0,0 +1,117 @@
+	return transactionsData, nil
+}
+
+const unexpectedHashTemplate = "missmatch on transaction data for batch num %d. Expected hash %s, actual hash: %s"
```
I would move global constants and variables to the top of the file.

Stefan-Ethernal commented on dataavailability/dataavailability.go:
```diff
@@ -0,0 +1,117 @@
+
+// New creates a DataAvailability instance
+func New(
+	IsTrustedSequencer bool,
+	backend DABackender,
+	state stateInterface,
+	zkEVMClient syncinterfaces.ZKEVMClientInterface,
+) (*DataAvailability, error) {
+	da := &DataAvailability{
+		isTrustedSequencer: IsTrustedSequencer,
```
```suggestion
	isTrustedSequencer bool,
	backend DABackender,
	state stateInterface,
	zkEVMClient syncinterfaces.ZKEVMClientInterface,
) (*DataAvailability, error) {
	da := &DataAvailability{
		isTrustedSequencer: isTrustedSequencer,
```

Stefan-Ethernal commented on dataavailability/datacommittee/datacommittee.go:
```diff
@@ -0,0 +1,292 @@
+// getCurrentDataCommitteeMembers return the currently registered data committee members
+func (d *DataCommitteeBackend) getCurrentDataCommitteeMembers() ([]DataCommitteeMember, error) {
+	members := []DataCommitteeMember{}
+	nMembers, err := d.dataCommitteeContract.GetAmountOfMembers(&bind.CallOpts{Pending: false})
+	if err != nil {
+		return nil, fmt.Errorf("error getting GetAmountOfMembers from L1 SC: %w", err)
+	}
+	for i := int64(0); i < nMembers.Int64(); i++ {
```
Minor optimization, providing initial capacity to the slice WDYT?

```suggestion
	nMembers, err := d.dataCommitteeContract.GetAmountOfMembers(&bind.CallOpts{Pending: false})
	if err != nil {
		return nil, fmt.Errorf("error getting GetAmountOfMembers from L1 SC: %w", err)
	}
	members := make([]DataCommitteeMember, 0, nMembers.Int64())
```

goran-ethernal commented on dataavailability/datacommittee/datacommittee.go:
```diff
@@ -0,0 +1,292 @@
+	return nil
+}
+
+const unexpectedHashTemplate = "missmatch on transaction data for batch num %d. Expected hash %s, actual hash: %s"
```
Move to the top of the file.

Stefan-Ethernal reviewed (approved):
Linter is complaining that there are some unused fields and functions in `cmd/policy.go`, although they are indeed used. Maybe we should try to suppress `unused` warning for that file?

Leaving some comments to consider.

Stefan-Ethernal reviewed (approved):
Linter is complaining that there are some unused fields and functions in `cmd/policy.go`, although they are indeed used. Maybe we should try to suppress `unused` warning for that file?

Leaving some comments to consider.

goran-ethernal commented on dataavailability/datacommittee/datacommittee.go:
```diff
@@ -0,0 +1,292 @@
+const unexpectedHashTemplate = "missmatch on transaction data for batch num %d. Expected hash %s, actual hash: %s"
+
+// GetData returns the data from the DAC. It checks that it matches with the expected hash
+func (d *DataCommitteeBackend) GetData(batchNum uint64, hash common.Hash) ([]byte, error) {
```
By looking at this `for` loop, seems there can be cases where it can run forever. Should we add some timeout to stop its execution after some time?

arnaubennassar replied:
the code is hard to follow, but should always break after reaching the initial committee member that has been queried

goran-ethernal commented on dataavailability/datacommittee/datacommittee.go:
```diff
@@ -0,0 +1,292 @@
+
+	// Collect signatures
+	msgs := []signatureMsg{}
+	var collectedSignatures uint64
```
```suggestion
	var (
	    collectedSignatures uint64
	    failedToCollect uint64
	 )
```

goran-ethernal commented on dataavailability/datacommittee/datacommittee.go:
```diff
@@ -0,0 +1,292 @@
+}
+
+func buildSignaturesAndAddrs(msgs signatureMsgs, members []DataCommitteeMember) []byte {
+	res := []byte{}
```
```suggestion
	res := make([]byte, 0, len(msgs)+len(members))
```

goran-ethernal commented on dataavailability/datacommittee/datacommittee_test.go:
```diff
@@ -0,0 +1,127 @@
+}
+
+// This function prepare the blockchain, the wallet with funds and deploy the smc
+func newTestingEnv() (
```
Pass `testing.T` to it, and set `t.Helper()` at the beginning of the function to mark it as test helper function.

goran-ethernal commented on dataavailability/datacommittee/datacommittee_test.go:
```diff
@@ -0,0 +1,127 @@
+
+// NewSimulatedEtherman creates an etherman that uses a simulated blockchain. It's important to notice that the ChainID of the auth
+// must be 1337. The address that holds the auth will have an initial balance of 10 ETH
+func newSimulatedDacman(auth *bind.TransactOpts) (
```
Pass `testing.T` to it, and set `t.Helper()` at the beginning of the function to mark it as test helper function.

goran-ethernal commented on etherman/etherman.go:
```diff
@@ -467,24 +754,82 @@ func (etherMan *Client) updateZkevmVersion(ctx context.Context, vLog types.Log,
+		return err
+	}
+
+	var gExitRoot GlobalExitRoot
```
```suggestion
	 gExitRoot := GlobalExitRoot {
	      MainnetExitRoot: globalExitRootL1InfoTree.MainnetExitRoot,
	      RollupExitRoot: globalExitRootL1InfoTree.RollupExitRoot,
	      BlockNumber: vLog.BlockNumber,
	      GlobalExitRoot: hash(globalExitRootL1InfoTree.MainnetExitRoot, globalExitRootL1InfoTree.RollupExitRoot),
	 }
```

arnaubennassar replied:
that comes from upstream, I'm consciously choosing to reduce code diff over clean code

goran-ethernal commented on etherman/etherman_test.go:
```diff
@@ -193,21 +178,31 @@ func TestSequencedBatchesEvent(t *testing.T) {
-	assert.Equal(t, auth.From, blocks[3].SequencedBatches[0][0].SequencerAddr)
-	assert.Equal(t, currentBlock.Time(), blocks[3].SequencedBatches[0][0].MinForcedTimestamp)
-	assert.Equal(t, 0, order[blocks[3].BlockHash][0].Pos)
+	// t.Log("Blocks: ", blocks)
```
Should we remove this commented code?

arnaubennassar replied:
comes from upstream

ARR552 committed:
Feature/aggregator update etrog batch (#3117)

* wip

* Fix aggregator

* change name to UpgradeEtrogBatchNumber

* doc

vcastellm commented on test/scripts/sendForcedBatch/main.go:
```diff
@@ -105,34 +119,34 @@ func sendForcedBatches(cliCtx *cli.Context) error {
-			return err
-		}
-	}
+	// // Allow forced batches in smart contract if disallowed
```
Is this a leftover?

arnaubennassar replied:
comes from upstream

arnaubennassar replied:
I had to upgrade contracts beyond the version that was used in the commit I'm pulling changes from upstream. This made me comment a bunch of code related to forced batches. In the next pulling this code should be uncommented with some chenges

arnaubennassar committed:
Add reuqests from PR

agnusmor committed:
Add L2 block sequential process. Execute empty L2 block. Add tracking number to L2 block. Several refactors (#3118)

* add L2 block sequential process. Execute empty L2 block. Add tracking number to L2 block. Several refactors

* fix finalizer UT

arnaubennassar committed:
Fix conflicts

arnaubennassar committed:
Upgrade contracts and fix compilation errors

arnaubennassar committed:
Update L1 docker image for testing

arnaubennassar committed:
Fix UTs

arnaubennassar committed:
Update diff docs

arnaubennassar committed:
Update config docs

vcastellm commented on dataavailability/dataavailability.go:
```diff
@@ -0,0 +1,117 @@
+			log.Info("trying to get data from trusted sequencer")
+			data, err := d.getDataFromTrustedSequencer(batchNum, expectedTransactionsHash)
+			if err != nil {
+				log.Error("failed to get data from trusted sequencer: %w", err)
```
Are these really errors or warnings?

arnaubennassar replied:
will lower log level in next PR

vcastellm commented on sequencesender/sequencesender.go:
```diff
@@ -35,18 +34,18 @@ type SequenceSender struct {
 	etherman     etherman
 	eventLog     *event.EventLog
-	privKey      *ecdsa.PrivateKey
+	da           dataAbilitier
```
`dataEnabler` better?

vcastellm reviewed (approved):
Aside from minor comments  LGTM

arnaubennassar committed:
WIP

arnaubennassar committed:
Fix linter

arnaubennassar committed:
Remove unnecessary GHA

arnaubennassar committed:
Update diff html

arnaubennassar committed:
Fix linter

arnaubennassar committed:
Expand explanation of the code diff

arnaubennassar committed:
minor config docs fixes

arnaubennassar committed:
Merge pull request #61 from 0xPolygon/pullupstream/v0.5.0-RC23

Pullupstream/v0.5.0 rc23
